File: backend/chat/chat_service.py
import google.generativeai as genai
API_KEY = "AIzaSyD3zJ_ihXbUbwOmFEmAy0x8PKMeiGokl4Y"


genai.configure(api_key=API_KEY)

model = genai.GenerativeModel('gemini-pro')
chat = model.start_chat(history=[])

while True:
    user_input = input("You: ")
    if user_input == "exit":
        break
    response = chat.send_message(user_input)
    print('\n')
    print("Gemini: " + response.text)
    print('\n')

File: src/services/chatService.js
// src/services/chatService.js
import google from 'google.generativeai';
import { get, ref } from './firebase';

export const getGeminiResponse = async (userId, promptKey, additionalInfo = '') => {
    const userSnapshot = await get(ref(database, 'users/' + userId));
    const user = userSnapshot.val();
    const apiKey = user.apiKey;

    google.generativeai.configure({ apiKey });

    const model = google.generativeai.GenerativeModel('gemini-pro');
    const chat = model.start_chat({ history: [] });

    const prompt = require('../models/prompts.json')[promptKey];
    const response = await chat.send_message(`${prompt} ${additionalInfo}`);
    return response.text;
};
